mscore/pianoroll/pianoview.cpp

```diff
@@ -881,39 +881,60 @@ void PianoView::dragSelectionNoteGroup() {
 //   addNote
 //---------------------------------------------------------
 
-void PianoView::addNote(Fraction startTick, Fraction frac, int pitch, int track, bool command)
+void PianoView::addNote(Fraction startTick, Fraction duration, int pitch, int track, bool command)
       {
       NoteVal nv(pitch);
 
       Score* score = _staff->score();
       Segment* seg = score->tick2segment(startTick);
       score->expandVoice(seg, track);
 
-      ChordRest* e = score->findCR(startTick, track);
-      if (e && !e->tuplet() && _tuplet == 1) {
-            //Ignore tuplets
+      ChordRest* curCr = score->findCR(startTick, track);
+      if (curCr && !curCr->tuplet() && _tuplet == 1) {
+            //Tuplets not handled yet
             if (command)
                   score->startCmd();
 
             ChordRest* cr0 = nullptr;
             ChordRest* cr1 = nullptr;
 
-            //Default to quarter note if faction is invalid
-            if (!frac.isValid() || frac.isZero())
-                  frac.set(1, 4);
+            //Default to quarter note if fraction is invalid
+            if (!duration.isValid() || duration.isZero())
+                  duration.set(1, 4);
 
-            if (cutChordRest(e, track, startTick, cr0, cr1)) {
-                  score->setNoteRest(cr1->segment(), track, nv, frac);
-                  }
-            else if (cr0) {
-                  if (cr0->isChord() && cr0->ticks() == frac) {
-                        Chord* ch = toChord(cr0);
+            if (startTick > curCr->tick())
+                  //Cut at the start of existing chord rest
+                  cutChordRest(curCr, track, startTick, cr0, cr1);
```

Comment:
To show that really no braces are needed, I'd rather have this command on the same line as the statemwnt